packages/test/src/test-schedules.ts

```diff
@@ -542,6 +542,87 @@ if (RUN_INTEGRATION_TESTS) {
     }
   });
 
+  test.serial('Can list Schedules with a query string', async (t) => {
+    const { client } = t.context;
+
+    const groupId = randomUUID();
+    const createdScheduleHandlesPromises = [];
+    const expectedIds: string[] = [];
+    for(let i = 0; i < 4; i++) {
+      const scheduleId = `test-query-${groupId}-${i + 1}`
+      if(i < 2) {
```

Comment:
Duplicating the whole block force one to be more attentive to figure out how the branches differ.

Instead, I'd just initialize the `searchAttributes` variable first (either undefined/empty or containing the custom search attribute) and add the schedule id to the list of expected schedules at the same time.

Something like:

```
const searchAttributes: SearchAttributes = {};
if (i < 2) {
  searchAttributes['CustomKeywordField'] = [ ... ];
  expectedIds.push(scheduleId);
}
```

Reply:
Yes @mjameswh Correcting it right away. Thank you for correcting.